xbmc/CueDocument.cpp

```diff
@@ -194,9 +194,41 @@ void CCueDocument::GetSongs(VECSONGS &songs)
 {
   for (size_t i = 0; i < m_tracks.size(); ++i)
   {
-    CSong song;
-    GetSong(i, song);
-    songs.push_back(song);
+    CSong aSong;
+    const CCueTrack& track = m_tracks[i];
+    //Pass artist to MusicInfoTag object by setting artist description string only.
+    //Artist credits not used during loading from cue sheet.
+    if ((track.strArtist.length() == 0) && (m_strArtist.length() > 0))
+      aSong.strArtistDesc = m_strArtist;
+    else
+      aSong.strArtistDesc = track.strArtist;
+    //Pass album artist to MusicInfoTag object by setting album artist vector. 
+    aSong.SetAlbumArtist(StringUtils::Split(m_strArtist, g_advancedSettings.m_musicItemSeparator));
+    aSong.strAlbum = m_strAlbum;
+    aSong.genre = StringUtils::Split(m_strGenre, g_advancedSettings.m_musicItemSeparator);
+    aSong.iYear = m_iYear;
+    aSong.iTrack = track.iTrackNumber;
+    if (m_iDiscNumber > 0)
+      aSong.iTrack |= (m_iDiscNumber << 16); // see CMusicInfoTag::GetDiscNumber()
+    if (track.strTitle.length() == 0) // No track information for this track!
+      aSong.strTitle = StringUtils::Format("Track %2d", track.iTrackNumber);
+    else
+      aSong.strTitle = track.strTitle;
+    aSong.strFileName = track.strFile;
+    aSong.iStartOffset = track.iStartTime;
+    aSong.iEndOffset = track.iEndTime;
+    if (aSong.iEndOffset)
+      aSong.iDuration = (aSong.iEndOffset - aSong.iStartOffset + 37) / 75;
```

Comment:
I didn't design this just rearranged the code, so I don't know why it is like this!